src/main/java/org/elmlang/intellijplugin/features/ElmFoldingBuilder.java

```diff
@@ -0,0 +1,36 @@
+package org.elmlang.intellijplugin.features;
+
+import com.intellij.lang.ASTNode;
+import com.intellij.lang.folding.FoldingBuilderEx;
+import com.intellij.lang.folding.FoldingDescriptor;
+import com.intellij.openapi.editor.Document;
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.util.PsiTreeUtil;
+import org.elmlang.intellijplugin.psi.ElmFoldable;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.stream.Collectors;
+
+public class ElmFoldingBuilder extends FoldingBuilderEx {
+
+    @NotNull
+    @Override
+    public FoldingDescriptor[] buildFoldRegions(@NotNull PsiElement root, @NotNull Document document, boolean quick) {
+        return PsiTreeUtil.findChildrenOfType(root, ElmFoldable.class).stream()
+                .map(ElmFoldable::getFoldingDescriptor)
+                .collect(Collectors.toList()).toArray(new FoldingDescriptor[0]);
+    }
+
+    @Nullable
+    @Override
+    public String getPlaceholderText(@NotNull ASTNode node) {
+        return node.getPsi(ElmFoldable.class).getFoldingPlaceholderText();
+    }
+
+    @Override
+    public boolean isCollapsedByDefault(@NotNull ASTNode node) {
+        return false;
```

Comment:
This is a sensible default, but we should probably collapse imports by default. This would match Java behavior, which presumably many IntelliJ users expect.